web-api/src/ingestion/routes.rs

```diff
@@ -184,26 +182,33 @@ impl NewIsCandidate {
 }
 
 async fn validate_document_properties(
+    document_id: &DocumentId,
     properties: impl IntoIterator<Item = (String, Value)>,
-    storage: &impl storage::Size,
+    storage: &(impl storage::Size + storage::IndexedProperties),
 ) -> Result<DocumentProperties, Error> {
     let properties = properties
         .into_iter()
-        .map(|(id, property)| Ok::<_, Error>((id.try_into()?, property.try_into()?)))
-        .try_collect::<_, HashMap<_, DocumentProperty>, _>()?;
-    if let Some(publication_date) = properties.get("publication_date") {
-        let Some(publication_date) = publication_date.as_str() else {
-            return Err(InvalidDocumentProperty { value: publication_date.clone().into() }.into());
-        };
-        DateTime::parse_from_rfc3339(publication_date)?;
+        .map(|(id, property)| {
+            let property_id = DocumentPropertyId::try_from(id)?;
```

Comment:
:art: we should make the name unambiguous from the `document_id`
```suggestion
        .map(|(property_id, property)| {
            let property_id = DocumentPropertyId::try_from(property_id)?;
```